### PR TITLE
fix(api-client): Always store the cookie

### PR DESCRIPTION
**Pull request by ffflorian**
Open questions:

* Should we really always save the cookie?
* How can we resolve the `instanceof` problem?

### CODE REVIEW

File: packages/api-client/src/main/shims/node/cookie.ts [hunk 1]
```diff
@@ -24,18 +24,16 @@ import {Cookie as ToughCookie} from 'tough-cookie';
...
+}
 
-const loadExistingCookie = async (engine: CRUDEngine): Promise<Cookie> => {
+const loadExistingCookie = (engine: CRUDEngine): Promise<Cookie> => {
   return engine
     .read<PersistedCookie>(AUTH_TABLE_NAME, AUTH_COOKIE_KEY)
-    .catch(error => {
-      if (error instanceof StoreEngineError.RecordNotFoundError) {
+    .catch((error: Error) => {
+      if (error instanceof StoreEngineError.RecordNotFoundError || error.constructor.name === 'RecordNotFoundError') {
```
Comment (bennyn) [inline packages/api-client/src/main/shims/node/cookie.ts]: When `error instanceof StoreEngineError.RecordNotFoundError` is not working at all then we should always take `error.constructor.name === 'RecordNotFoundError'`.

Comment (ffflorian) [reply packages/api-client/src/main/shims/node/cookie.ts]: Actually we do the same in Account.ts:
https://github.com/wireapp/wire-web-packages/blob/3d2effa274e186d0c54b0c969f20c2b64b1736da/packages/core/src/main/Account.ts#L141-L146

So I vote for checking both with `instanceof` and `error.constructor.name`.

Comment (bennyn) [reply packages/api-client/src/main/shims/node/cookie.ts]: But there one check should be sufficient, no?

**Comment by bennyn** [comment]
@ffflorian No downsides that I know of.

**Pull request merged**